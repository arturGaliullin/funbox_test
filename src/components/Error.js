import React, { Component } from 'react';

export default class Error extends Component {
    render() {
        return (
            <div className='editor__error '>
                <p>{this.props.searchError ? this.props.searchError : null}</p>
                <p>{this.props.placemarkError ? this.props.placemarkError : null}</p>
            </div>
        )
    }
}