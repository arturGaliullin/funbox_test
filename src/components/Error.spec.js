import React from 'react';
import { shallow } from 'enzyme';
import Error from './Error';

const props = {
    searchError: false,
    placemarkError: false
};

describe('Error', () => {
    it('should render with props', () => {
        const component = shallow(<Error {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('should render with searchError is "Произошла ошибка при поиске, повторите позже."', () => {
        props.searchError = 'Произошла ошибка при поиске, повторите позже.';
        const component = shallow(<Error {...props} />);
        expect(component.find('p').first().text()).toEqual('Произошла ошибка при поиске, повторите позже.');
    });

    it('should render with placemarkError is "Произошла ошибка при загрузке данных метки, повторите позже."', () => {
        props.placemarkError = 'Произошла ошибка при загрузке данных метки, повторите позже.';
        const component = shallow(<Error {...props} />);
        expect(component.find('p').last().text()).toEqual('Произошла ошибка при загрузке данных метки, повторите позже.');
    });

});