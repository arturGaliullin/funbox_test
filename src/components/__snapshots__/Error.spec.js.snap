// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    placemarkError={false}
    searchError={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p />,
        <p />,
      ],
      "className": "editor__error ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p />,
          <p />,
        ],
        "className": "editor__error ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
